third-party/realdds/include/realdds/dds-device.h

```diff
@@ -50,13 +50,13 @@ class dds_device
 
     size_t num_of_sensors() const;
```

Comment:
If we're going away from a sensor-centric DDS communication towards stream-centric, why is the sensor API still present?

Reply:
I think the difference between `stream` and `profile` was unclear to me, I used them to describe the same thing
From now on I will use `stream` for a single kind of images (IR, color etc..) and `profile` for the different resolution and fps variations a `stream` can have.
`sensor` will not be used, but `streams` can be grouped together by name

Reply:
So this will turn into `n_streams()`?

Reply:
Turned into `num_of_streams` and `num_of_stream_groups`

Reply:
Why `num_of_stream_groups`? Is there a need?
I have to say I'm not a fan of `num` instead of `number`. If we want to shorten, `n_streams` is fairly well understood and an agreed shorthand for "number of"

Reply:
I don't like how we replaced all the "sensor" references with "stream_group".
I don't think we should provide **any** foreach, n_, etc. functions for the stream_group. This is a libRS interpretation. We should stick to streams and that's it.

Reply:
@maloel How do you suggest the client will know how to attach certain stream types to a `sw_sensor` ?

Reply:
Using group name stored in dds_stream objects to attach to `software_sensor`